cmplibrary/src/main/java/com/sourcepoint/cmplibrary/campaign/CampaignManager.kt

```diff
@@ -650,4 +652,45 @@ private class CampaignManagerImpl(
             ccpaCS = ccpaConsentStatus
         )
     }
+
+    override val isGdprExpired: Boolean
+        get() {
+
+            val formatter = SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ss.SSS'Z'", Locale.getDefault())
+            val gdprExpirationDate = dataStorage.gdprExpirationDate?.let { formatter.parse(it) } ?: return false
+            val currentDate = Date()
+            val isGdprExpired = currentDate.after(gdprExpirationDate)
+
+            logger?.computation(
+                tag = "Expiration Date",
+                msg = """
+                isGdprExpired[$isGdprExpired] 
+                """.trimIndent()
+            )
+
+            return isGdprExpired
+        }
+
+    override val isCcpaExpired: Boolean
+        get() {
+
+            val formatter = SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ss.SSS'Z'", Locale.getDefault())
+            val ccpaExpirationDate = dataStorage.ccpaExpirationDate?.let { formatter.parse(it) } ?: return false
```

Comment:
same here

Reply:
like before

Reply:
@carmelo-iriti so what's the solution? if formatter.parse(it) throws an exception, the ccpa consent will never expire?